### PR TITLE
Update common axis when slicing NDCubeSequence.

### LINKED ISSUE #247: NDCubeSequence._common_axis not altered by slicing
### Description

```NDCubeSequence._common_axis``` designates the data axis of the ```NDCube```s in the sequence that are common.  This data axis can change when the ```NDCubeSequence``` is sliced.  Therefore, ```NDCubeSequence._common_axis``` should update to reflect this.  However, this does not happen.

### Expected behavior

Say ```my_sequence``` is an ```NDCubeSequence``` with a ```common_axis``` of ```1```.
```python
>>> my_sequence._common_axis
1
```
Slicing away preceding axis should decrement common axis.
```python
>>> my_sequence[:, 0]._common_axis
0
```
Slicing away common axis should set common axis to None.
```python
>>> my_sequence[:, :, 0]._common_axis
None
```
Slicing away axis greater than common axis should have no effect.
```python
>>> my_sequence[:, :, :, 0]._common_axis
1
```

### Actual behavior

```my_sequence._common_axis``` never changes.
```python
>>> my_sequence._common_axis
1
>>> my_sequence[:, 0]._common_axis
1
>>> my_sequence[:, :, 0]._common_axis
1
>>> my_sequence[:, :, :, 0]._common_axis
1
```

### Steps to Reproduce

Step 1: Make an ```NDCubeSequence```
```python
>>> import numpy as np 
>>> data0 = np.ones((3, 4, 5)) 
>>> data1 = data0 * 2 
>>> data2 = data1 * 2 

>>> # Define WCS object for all cubes. 
>>> import astropy.wcs 
>>> wcs_input_dict = { 
... 'CTYPE1': 'WAVE    ', 'CUNIT1': 'Angstrom', 'CDELT1': 0.2, 'CRPIX1': 0, 'CRVAL1': 10, 'NAXIS1': 5, 
... 'CTYPE2': 'HPLT-TAN', 'CUNIT2': 'deg', 'CDELT2': 0.5, 'CRPIX2': 2, 'CRVAL2': 0.5, 'NAXIS2': 4, 
... 'CTYPE3': 'HPLN-TAN', 'CUNIT3': 'deg', 'CDELT3': 0.4, 'CRPIX3': 2, 'CRVAL3': 1, 'NAXIS3': 3} 
>>> input_wcs = astropy.wcs.WCS(wcs_input_dict)

>>> from ndcube import NDCube 
>>> from ndcube import NDCubeSequence 
>>> my_cube0 = NDCube(data0, input_wcs)
>>> my_cube1 = NDCube(data1, input_wcs)
>>> my_cube2 = NDCube(data2, input_wcs)

>>> my_sequence = NDCubeSequence([my_cube0, my_cube1, my_cube2], common_axis=1)
```

Step 2: Slicing preceding axis should make common axis iterate by -1 (or -(number of preceding axes sliced away)).  But it remains unchanged.
```
>>> my_sequence0._common_axis
1
>>> my_sequence0[:, 0]._common_axis  # Should return 0
1
```

Step 3: Slicing preceding axis should make common axis ```None```.  But it remains unchanged.
```
>>> my_sequence0[:, :, 0]._common_axis # Should return None
1
```

### System Details

  - ndcube Version: 1.3

### PR DESCRIPTION
**Pull request by DanRyanIrish**
This PR fixes #247

### CODE REVIEW

File: ndcube/ndcube_sequence.py [hunk 1]
```diff
@@ -87,25 +87,24 @@ def cube_like_array_axis_physical_types(self):
     def __getitem__(self, item):
         if isinstance(item, numbers.Integral):
             return self.data[item]
+        result = copy.deepcopy(self)
```
Comment (Cadair) [inline ndcube/ndcube_sequence.py:90]:
```suggestion
        result = type(self)([])
        result.meta = copy.deepcopy(self.meta)
        result._common_axis = self._common_axis
```

File: ndcube/ndcube_sequence.py [hunk 2]
```diff
@@ -87,25 +87,24 @@ def cube_like_array_axis_physical_types(self):
...
         else:
-            result = copy.deepcopy(self)
-            if isinstance(item, slice):
-                result.data = self.data[item]
+            if isinstance(item[0], numbers.Integral):
+                result = result.data[item[0]][item[1:]]
             else:
-                if isinstance(item[0], numbers.Integral):
-                    result = result.data[item[0]][item[1:]]
+                result.data = [cube[item[1:]] for cube in result.data[item[0]]]
```
Comment (Cadair) [inline ndcube/ndcube_sequence.py:97]:
```suggestion
                result.data = [cube[item[1:]] for cube in self.data[item[0]]]
```

**Review by Cadair — APPROVED** [review]
Looks good pending deepcopy changes

**Comment by DanRyanIrish** [comment]
Add this to the isort section of `setup.cfg` to exclude isort from running on the .rst files and make codestyle pass

```skip_glob = *.rst```

**Pull request merged**